### DISCUSSION TITLE
strange behaviour in azur app services/containers + opentelemetry-js

### DISCUSSION THREAD

**Discussion by brechr** [open-telemetry/community#1095] [Community Discussions]
Hey everyone,

we're running a certain node/typescript service in an Azure App Service for Containers on a Linux Machine and i can't get OTel to record the spans i want it to record inside of this certain enviornment.

What I can tell: 
- This worked until revecently and suddenly stopped working. I was thinking it continued working later (without reason) but maybe I was seeing things
- the api.diag logger prints "Recording is off, propagating context in a non-recording span" for most spans that i've deinfed/created and wanted to export
- there is exactly one HTTP Request that is recorded, it is '/robots933456.txt' which is kind of a default healththeck/ping request from azure afaik and possibly most importantly: If I request this file from 'the outside' the request isn't recorded.
- everything works fine in local environments or local docker, with different exporters and span processors.. doesn't seem to matter at all. 

I'm at the brink of taking the whole app apart/remote debugging the hell out of it but because it's driving me insane, especially the fact that this just stopped working out of nowhere and I don't understand why. But before I do this I figured I could ask here (hoping that it's the right place):

Does anyone have an idea what is going wrong here and how to fix this? If this is the wrong place to ask, where should I go? directly to opentelemetry-js? 

Thanks a lot!
Christian

**Comment by habmic** [comment]
Hi @brechr I think the best would be to ask it in the CNCF slack channel of #otel-js as it mostly debugging / troubleshooting issue or you can try to open an issue in the js repo

BTW, the first thing to do would be to turn on otel logs:
```

import {
    diag,
    DiagConsoleLogger,
    DiagLogLevel,
} from '@opentelemetry/api';
diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.ALL);
```

**Reply by brechr** [reply]
hi @habmic, thanks will try it there.

I already enabled the otel logs, that's where the 

> api.diag logger prints "Recording is off, propagating context in a non-recording span"

is coming from. I'll mark your reply as answer and will try the issue at the js repo way. :)

Thanks a lot and have a nice day
